extensions/vertx-web/deployment/src/main/java/io/quarkus/vertx/web/deployment/VertxWebProcessor.java

```diff
@@ -574,7 +602,32 @@ void implementInvoke(HandlerDescriptor descriptor, BeanInfo bean, MethodInfo met
                 defaultProduces == null ? invoke.loadNull() : invoke.load(defaultProduces));
 
         // Invoke the business method handler
-        ResultHandle res = invoke.invokeVirtualMethod(methodDescriptor, beanInstanceHandle, paramHandles);
+        AssignableResultHandle res;
+        if (descriptor.isReturningUni()) {
+            res = invoke.createVariable(Uni.class);
+        } else if (descriptor.isReturningMulti()) {
+            res = invoke.createVariable(Multi.class);
+        } else {
+            res = invoke.createVariable(Object.class);
+        }
+        invoke.assign(res, invoke.loadNull());
+        if (!descriptor.requireValidation()) {
+            ResultHandle value = invoke.invokeVirtualMethod(methodDescriptor, beanInstanceHandle, paramHandles);
+            if (value != null) {
+                invoke.assign(res, value);
+            }
+        } else {
+            TryBlock block = invoke.tryBlock();
+            ResultHandle value = block.invokeVirtualMethod(methodDescriptor, beanInstanceHandle, paramHandles);
+            if (value != null) {
+                block.assign(res, value);
+            }
+            CatchBlockCreator caught = block.addCatch("javax.validation.ConstraintViolationException");
```

Comment:
I'd suggest to use constants for repeating strings...

Reply:
Fixed